packages/udt/src/udt/index.ts

```diff
@@ -274,33 +367,514 @@ export class Udt extends ssri.Trait {
+        return acc + ccc.udtBalanceFrom(outputData);
+      },
+      ccc.numFrom(0),
+    );
+  }
+
+  /**
+   * Calculates the total UDT balance from all outputs in a transaction.
+   * This method examines each output cell and sums up the UDT amounts
+   * for cells that have this UDT's type script.
+   *
+   * @param txLike - The transaction to analyze
+   * @returns The total UDT balance from all outputs
+   *
+   * @example
+   * ```typescript
+   * const udt = new Udt(codeOutPoint, scriptConfig);
+   * const tx = ccc.Transaction.from({
+   *   outputs: [
+   *     { lock: recipientLock, type: udt.script },
+   *     { lock: changeLock, type: udt.script }
+   *   ],
+   *   outputsData: [
+   *     ccc.numLeToBytes(1000, 16), // 1000 UDT to recipient
+   *     ccc.numLeToBytes(500, 16)   // 500 UDT as change
+   *   ]
+   * });
+   *
+   * const outputBalance = udt.getOutputsBalance(tx);
+   * console.log(`Total UDT output balance: ${outputBalance}`); // 1500
+   * ```
+   *
+   * @remarks
+   * This method only counts outputs that have the same type script as this UDT instance.
+   * Outputs without a type script or with different type scripts are ignored.
+   * This is a synchronous method since output data is already available in the transaction.
+   */
+  getOutputsBalance(txLike: ccc.TransactionLike): ccc.Num {
+    const tx = ccc.Transaction.from(txLike);
+    return tx.outputs.reduce((acc, output, i) => {
+      if (!this.isUdtCell(output, tx.outputsData[i])) {
+        return acc;
+      }
+
+      return acc + ccc.udtBalanceFrom(tx.outputsData[i]);
+    }, ccc.numFrom(0));
+  }
+
+  /**
+   * Calculates the net UDT balance that would be burned (destroyed) in a transaction.
+   * This is the difference between the total UDT balance in inputs and outputs.
+   * A positive value indicates UDT tokens are being burned, while a negative value
+   * indicates more UDT is being created than consumed (which may require minting permissions).
+   *
+   * @param txLike - The transaction to analyze
+   * @param client - The client to fetch input cell data
+   * @returns A promise resolving to the net UDT balance burned (inputs - outputs)
+   *
+   * @example
+   * ```typescript
+   * const udt = new Udt(codeOutPoint, scriptConfig);
+   * const tx = ccc.Transaction.from(existingTransaction);
+   *
+   * const burned = await udt.getBalanceBurned(tx, client);
+   * if (burned > 0) {
+   *   console.log(`${burned} UDT tokens will be burned`);
+   * } else if (burned < 0) {
+   *   console.log(`${-burned} UDT tokens will be created`);
+   * } else {
+   *   console.log('UDT balance is conserved');
+   * }
+   * ```
+   *
+   * @remarks
+   * This method is useful for:
+   * - Validating transaction balance conservation
+   * - Calculating how much UDT is being destroyed in burn operations
+   * - Detecting minting operations (negative burned balance)
+   * - Ensuring sufficient UDT inputs are provided for transfers
+   */
+  async getBalanceBurned(
```

Comment:
I would call it getBalanceDelta, cause this delta may indeed be positive or negative

Reply:
Also if we add the CKB to the input and output methods, also this one should return its delta

Reply:
Once in a while, I name it `balanceDiff`. I think delta has the same problem as diff, in that we can't intuitively understand whether it represents input minus output or vice versa.

Reply:
Delta usually is defined Output minus Input, differently from diff, which has no such connotation. That said, Burned is surely more colorful and probably intuitive, maybe it'll cause also more raised eyebrows, but nothing a good TypeDoc can't fix